### ISSUE TITLE
type (MUST be a single value)

### ISSUE THREAD

**Issue by trwnh** [steve-bate/activitypub-mincore#3]
if your goal is to be compatible with most fedi impls, this is not enough. you specifically need to declare a type that is one of the five types in AS2 or else Mastodon/Pleroma/Misskey/etc will reject it

however: is that indeed the aim, or is the aim to be minimal *activitypub*? i would also like to make the point that extensions will almost certainly have multiple types, so this is something that ought to be considered at least on the consuming side, even if not for the single actor being produced. (per https://matrix.to/#/!uHqAjmOtrLtidOiczC:matrix.org/$LKdlBM3fEStSux5SDXFSwnGATPNtMLFssS99J2FB7q0?via=matrix.org&via=mozilla.org&via=beeper.com and related discussion above that point)

**Comment by steve-bate** [comment]
Yes, this is focused on minimal *activitypub*, not a minimal Mastodon-interoperable set of features. The goal is to experiment with an approach where we define a minimal core set of functionality and interoperability profiles (also sometimes called "conformance" or "compliance" profiles) layered on top of that. These conformance profiles can relax and extend the minimal core for specific purposes or social web domains, like microblogging.

Examples:
* AP Minimal Core (AP Core)
* Interoperability Profiles (layered on AP Core)
  * Microblogging (Mastodon-flavored AP)
  * Image sharing (probably based on PixelFed)
  * Forums
  * Marketplaces
  * Long-form Writing
  * Internet/Web of Things
  * Distributed Moderation (cross-cutting)
  * Activity Streaming (SSE, Websockets, etc. - cross-cutting)
  * Full Linked Data Support (cross-cutting)
  * . . .
 


In practice, I wouldn't expect any nontrivial server to *only* implement the AP Core. They'd implement one or more interoperability profiles. The work to define those has not been done yet. 

This would be an alternative to the current monolithic AP/AS2 specification that tries to be the union of many different interoperability scenarios and, ironically, insufficient for all of those scenarios in practice.

I don't think of the AP Core itself as an interoperability profile, but it could be viewed that way. If so, then an AP Core conformant server could reject any activities with multivalued types as nonconformant (to the profile it implemented).
